AirCasting/Utils/Connection/BluetoothConnection.swift

```diff
@@ -0,0 +1,53 @@
+import SwiftUI
+import CoreData
+import Resolver
+
+final class BluetoothConnectionProtector: Connectable, SessionsFetchable {
+    
+    enum BluetoothConnectionProtectorError: Error {
+        case alreadyConnected
+        case readError(Error)
+    }
+    
+    @Injected private var persistenceController: PersistenceController
```

Comment:
if you opt to use `SessionsFetchable`, you won't need this (we will be moving to a higher abstraction that hides all that CoreData complexity)